### PR TITLE
feat: support arbitary boto3 s3 config options in s3 backend

### PR DESCRIPTION
**Pull request by joel-aws**

### CODE REVIEW

File: src/bandersnatch_storage_plugins/s3.py [hunk 1]
```diff
@@ -123,6 +123,8 @@ class S3Storage(StoragePlugin):
     PATH_BACKEND = S3Path
     resource = None
     UPLOAD_TIME_METADATA_KEY = "uploaded-at"
+    CONFIG_PREFIX = "config_param_"
```
Comment (cooperlees) [inline src/bandersnatch_storage_plugins/s3.py:126]:
```suggestion
    BOTO_CONFIG_PREFIX = "config_param_"
```
Maybe to make it clearer it's use

**Review by cooperlees — CHANGES_REQUESTED** [review]
Hi and thanks for this. I am sure it will be helpful for S3 bandersnatch users.

Is there any easy way we can use our [minio unit/integration tests](https://github.com/pypa/bandersnatch/blob/main/src/bandersnatch/tests/plugins/test_storage_plugin_s3.py) to make use of any of these config params to see at least one work? I feel this will help with future API changes etc. etc.

File: src/bandersnatch_storage_plugins/s3.py (hunk 1 above)
Comment (joel-aws) [reply src/bandersnatch_storage_plugins/s3.py:126]: Agreed -- that's more clear. Will update.

**Comment by joel-aws** [comment]
@cooperlees -- added a test that uses minio.

I also refactored `s3.py` to apply the boto3 configuration to all instantiation of `S3Path`. Though, when `.s3keep` objects are created, the configuration is *not* applied (couldn't figure that one out). Overall, I don't love my implementation, so I welcome suggestions.

**Review by cooperlees — APPROVED** [review]
I don't have suggestions no how to make this any cleaner. So I am happy to merge if you are.

> Though, when .s3keep objects are created, the configuration is not applied

Is there any open issue or is it worth opening an issue here to see if there is a bug or we're doing something wrong here?

If you fix the CHANGES entry I'll merge and if it benefits you cut a new release with the feature.

Thanks for the contribution! (I am no s3 guru tho)

**Comment by cooperlees** [comment]
@joel-aws Happy for this to be merged?

**Comment by joel-aws** [comment]
@cooperlees LGTM!

**Pull request merged**